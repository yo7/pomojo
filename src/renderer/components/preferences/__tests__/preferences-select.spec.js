import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import PreferencesSelect from '../preferences-select.vue'

describe('PreferencesSelect', () => {
  let wrapper
  const onChanged = jest.fn()
  beforeEach(() => {
    wrapper = shallow(PreferencesSelect, {propsData: {
      options: [1, 2, 3],
      current: 2,
      onChanged
    }})
  })

  it('has as many options as its props', () => {
    expect(wrapper.findAll('option').length).toBe(3)
  })

  it('call onChanged on select clicked', () => {
    const select = wrapper.find('.select')
    select.trigger('change')
    expect(onChanged).toHaveBeenCalled()
  })

  it('renders correctly', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) {
        throw new Error()
      }
      expect(str).toMatchSnapshot()
    })
  })
})
