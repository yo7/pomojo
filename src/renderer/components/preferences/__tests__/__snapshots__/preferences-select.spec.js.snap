// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencesSelect renders correctly 1`] = `
<label class="preferences-select"><select class="select"><option value="1">
      1
    </option><option value="2" selected="selected">
      2
    </option><option value="3">
      3
    </option></select></label>
`;
